### PR TITLE
Feature/dp mongo v2

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What
Migrate to use dp-mongo v2

### How to review
Review changes / check tests pass

### Who can review
Anyone

### CODE REVIEW

File: mongo/mongo.go [hunk 1]
```diff
@@ -65,27 +86,26 @@ func (m *Mongo) Checker(ctx context.Context, state *healthcheck.CheckState) erro
...
-		q.Sort("publish_date")
+		q.Sort(bson.D{{"publish_date", 1}})
 	}
 
-	totalCount, err := q.Count()
+	totalCount, err := q.Count(ctx)
 	if err != nil {
 		log.Error(ctx, "error getting count of collections from mongo db", err)
-		if err == mgo.ErrNotFound {
+		if dpMongoDriver.IsErrNoDocumentFound(err) {
```
Comment (s-xton) [inline mongo/mongo.go:108]: count will not return an error here

File: mongo/mongo.go [hunk 2]
```diff
@@ -94,17 +114,9 @@ func (m *Mongo) GetCollections(ctx context.Context, queryParams collections.Quer
...
-			if err != nil {
-				log.Error(ctx, "error closing iterator", err)
-			}
-		}()
-
-		if err := iter.All(&values); err != nil {
-			if err == mgo.ErrNotFound {
+		err = q.Skip(queryParams.Offset).Limit(queryParams.Limit).IterAll(ctx, &values)
+		if err != nil {
+			if dpMongoDriver.IsErrNoDocumentFound(err) {
```
Comment (s-xton) [inline mongo/mongo.go:119]: you will not receive this error you will get an empty array

File: mongo/mongo.go (hunk 1 above)
Comment (CarlHembrough) [reply mongo/mongo.go:108]: Not sure exactly what you mean by this - are you suggesting that count will not return a `ErrNoDocumentFound` error?

Comment (CarlHembrough) [reply mongo/mongo.go]: I have assumed this is the case and removed the specific error checking

File: mongo/mongo.go (hunk 2 above)
Comment (CarlHembrough) [reply mongo/mongo.go]: Removed check for `ErrNoDocumentFound`

**Review by s-xton — APPROVED** [review]

**Review by s-xton — APPROVED** [review]

**Pull request merged**